src/rendezvous/channels/ecdhV1.ts

```diff
@@ -0,0 +1,298 @@
+            ]);
+
+            return JSON.stringify({
+                iv: encodeBase64(iv),
+                ciphertext: encodeBase64(ciphertext),
+            });
+        }
+
+        if (!subtleCrypto) {
+            throw new Error('Neither Web Crypto nor Node.js crypto are available');
+        }
+
+        const iv = new Uint8Array(32);
+        window.crypto.getRandomValues(iv);
+
+        const encodedData = new TextEncoder().encode(data);
+
+        const ciphertext = await subtleCrypto.encrypt(
+            {
+                name: "AES-GCM",
+                iv,
+                tagLength: 128,
+            },
+            this.aesKey as CryptoKey,
+            encodedData,
+        );
+
+        return JSON.stringify({
+            iv: encodeBase64(iv),
+            ciphertext: encodeBase64(ciphertext),
+        });
+    }
+
+    public async send(data: any) {
+        if (!this.olmSAS) {
+            throw new Error('Channel closed');
+        }
+
+        const stringifiedData = JSON.stringify(data);
+
+        if (this.aesKey) {
+            await this.transport.send('application/json', await this.encrypt(stringifiedData));
+        } else {
+            await this.transport.send('application/json', stringifiedData);
+        }
+    }
+
+    private async decrypt({ iv, ciphertext }: { iv: string, ciphertext: string }): Promise<any> {
+        if (!ciphertext || !iv) {
+            throw new Error('Missing ciphertext and/or iv');
+        }
+
+        const ciphertextBytes = decodeBase64(ciphertext);
+
+        if (this.aesKey instanceof Uint8Array) {
+            const crypto = getCrypto();
+            // in contrast to Web Crypto API, Node's crypto needs the auth tag split off the cipher text
+            const ciphertextOnly = ciphertextBytes.slice(0, ciphertextBytes.length - 16);
+            const authTag = ciphertextBytes.slice(ciphertextBytes.length - 16);
+            const decipher = crypto.createDecipheriv(
+                "aes-256-gcm", this.aesKey as Uint8Array, decodeBase64(iv), { authTagLength: 16 },
+            );
+            decipher.setAuthTag(authTag);
+            return decipher.update(encodeBase64(ciphertextOnly), "base64", "utf-8") + decipher.final("utf-8");
+        }
+
+        if (!subtleCrypto) {
+            throw new Error('Neither Web Crypto nor Node.js crypto are available');
+        }
+
+        const plaintext = await subtleCrypto.decrypt(
+            {
+                name: "AES-GCM",
+                iv: decodeBase64(iv),
+                tagLength: 128,
+            },
+            this.aesKey as CryptoKey,
+            ciphertextBytes,
+        );
+
+        return new TextDecoder().decode(new Uint8Array(plaintext));
+    }
+
+    public async receive(): Promise<any> {
+        if (!this.olmSAS) {
+            throw new Error('Channel closed');
+        }
+
+        const data = await this.transport.receive();
+        if (!data) {
+            return data;
```

Comment:
Probably should return a fixed value (e.g. `null`?) to indicate that nothing was received, rather than returning the empty data as-is?

Reply:
Now undefined.